### ISSUE TITLE
Compatibility with `graphql-codegen` DocumentNodes (`Cannot read properties of undefined (reading 'some')`)

### ISSUE THREAD

**Issue by benjaminwaterlot** [appmotion/apollo-augmented-hooks#20]
Hello there 👋 🙂
      
Thanks a lot for open-sourcing this project - well-written and the "reduced" queries are a very useful feature for us.

One note: as-is, it's incompatible with `DocumentNode` generated by `graphql-codegen`, because empty `selection.directives`, `selection.variableDefinitions` and `directive.arguments` are **not present** in the generated `DocumentNode` instead of being empty arrays.

FYI, here's the `patch-package` diff that solves the problem for us (using `graphql-codegen` with the 'typescript-operations' and 'typed-document-node' plugins).

```diff
diff --git a/node_modules/apollo-augmented-hooks/lib/helpers/reducedQueries.js b/node_modules/apollo-augmented-hooks/lib/helpers/reducedQueries.js
index 3280589..ff65733 100644
--- a/node_modules/apollo-augmented-hooks/lib/helpers/reducedQueries.js
+++ b/node_modules/apollo-augmented-hooks/lib/helpers/reducedQueries.js
@@ -110,7 +110,7 @@ var filterSubSelections = function filterSubSelections(selections, cacheData, ca
     // local-only.
 
 
-    if (selection.directives.some(function (directive) {
+    if ((selection.directives || []).some(function (directive) {
       return directive.name.value === 'client';
     })) {
       return result;
@@ -200,9 +200,9 @@ var hasVariable = function hasVariable(selectionSet, variable) {
       return true;
     }
 
-    var isVariableInArguments = hasArgumentVariable(selection.arguments, variable);
-    var isVariableInDirectives = selection.directives.some(function (directive) {
-      return directive.arguments.some(function (_ref2) {
+    var isVariableInArguments = hasArgumentVariable(selection.arguments || [], variable);
+    var isVariableInDirectives = (selection.directives || []).some(function (directive) {
+      return (directive.arguments || []).some(function (_ref2) {
         var _value$name;
 
         var value = _ref2.value;
@@ -286,7 +286,7 @@ var makeReducedQueryAst = function makeReducedQueryAst(cache, queryAst, variable
         selectionSet: selectionSet,
         // Remove variable definitions that are no longer referenced anywhere in the selection
         // set.
-        variableDefinitions: definition.variableDefinitions.filter(function (_ref3) {
+        variableDefinitions: (definition.variableDefinitions || []).filter(function (_ref3) {
           var variable = _ref3.variable;
           return hasVariable(selectionSet, variable.name.value);
         })
```

Sorry for not submitting a clean PR - I'm missing time to go more in-depth but wanted to help unblocking other users of the lib. 

Note that this is the minimal set of changes that allows us to use `useQuery`, but I'm not sure it's exhaustive.

**Comment by mindnektar** [comment]
Hey there! Thanks a lot for this very clear and concise issue! I've implemented your changes and published them in v4.0.4. Glad to know this little package is useful to others, too. :)

**Comment by mindnektar** [comment]
You're very welcome!
